units/patch_intent.go

```diff
@@ -311,9 +312,28 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 		if buildVariant == "all" || buildVariant == "" {
 			continue
 		}
-		bv := project.FindBuildVariant(buildVariant)
-		if bv == nil {
-			return errors.Errorf("No such buildvariant: '%s'", buildVariant)
+		bvRegex, err := regexp.Compile(buildVariant)
+		if err != nil {
+			return errors.Wrapf(err, "unable to compile regex %s", buildVariant)
+		}
+		bvs := project.FindMatchingBuildVariants(bvRegex)
+		if len(bvs) == 0 {
+			return errors.Errorf("No such buildvariant matching '%s'", buildVariant)
+		}
+	}
+
+	// verify that all tasks exists
```

Comment:
Is this comment needed or can it be removed?

Reply:
removed